ipam/space/space.go

```diff
@@ -96,12 +100,19 @@ func (s *Space) Free(addr address.Address) error {
 	return nil
 }
 
-func (s *Space) biggestFreeRange() (int, address.Offset) {
+func (s *Space) biggestFreeRange(rangeStart, rangeEnd address.Address) (int, address.Offset) {
 	pos := -1
 	biggest := address.Offset(0)
 
 	for i := 0; i < len(s.free); i += 2 {
-		size := address.Subtract(s.free[i+1], s.free[i])
+		start, end := s.free[i], s.free[i+1]
+		if start < rangeStart {
+			start = rangeStart
+		}
+		if end > rangeEnd {
+			end = rangeEnd
+		}
```

Comment:
(Although GitHub is helpfully retaining many comments here, all of them relate to older versions of the code)

What I meant was, it takes four if-statements to get the correct answer while applying short-circuits, so I had omitted the short-circuits in one case.  I have extracted the logic into a helper function so being long-winded isn't such a problem.